lib/app/utils/startup_tasks.rb

```diff
@@ -0,0 +1,103 @@
+# frozen_string_literal: true
+
+require 'rubygems/package'
+
+module Inferno
+  module StartupTasks
+    class << self
+      def run
+        check_validator_availability
+        load_all_modules
+      end
+
+      def load_all_modules
+        Dir.glob(File.join(__dir__, '..', '..', 'modules', '*_module.yml')).each do |file|
+          module_metadata = YAML.load_file(file).deep_symbolize_keys
+          Module.new(module_metadata)
+
+          load_ig_in_validator(module_metadata) if external_validator? && module_metadata.key?(:resource_path)
+        end
+      end
+
+      def external_validator?
+        App::Endpoint.settings.resource_validator == 'external'
+      end
+
+      def validator_url
+        App::Endpoint.settings.external_resource_validator_url
+      end
+
+      def check_validator_availability
+        unless external_validator?
+          Inferno.logger.info 'Using internal validator'
+          return
+        end
+
+        loop do
+          Inferno.logger.info "Checking that validator is available at #{validator_url}"
+          validator_version = RestClient.get("#{validator_url}/version")
+          Inferno.logger.info "External validator version #{validator_version} is available"
+          break
+        rescue StandardError
+          Inferno.logger.error "Unable to reach validator at #{validator_url}"
+          sleep 1
+        end
+      end
+
+      def load_ig_in_validator(module_metadata)
+        module_name = module_metadata[:title].presence || module_metadata[:name]
+        unless package_json_present? module_metadata[:resource_path]
+          Inferno.logger.info "Skipping validator upload for '#{module_name}'--no package.json file."
+          return
+        end
+
+        Inferno.logger.info "Creating ig package for '#{module_name}'"
+        Tempfile.open([module_metadata[:resource_path], '.tar.gz']) do |file|
+          create_ig_zip(file, module_metadata)
+
+          begin
+            Inferno.logger.info "Posting IG for '#{module_name}' to validator"
+            RestClient.post("#{validator_url}/igs", File.read(file.path))
```

Comment:
From what I can tell, you are not using the response from this route for anything, right? I just wanted to make sure because in [inferno-community/fhir-validator-wrapper#16](https://github.com/inferno-community/fhir-validator-wrapper/pull/16), we recently changed the response for the `PUT` and `POST /igs` routes. Just wanted to give you a heads-up in case this might break some functionality in this PR.

Reply:
If it receives a 4xx/5xx response, then an error is raised, otherwise it's not doing anything with the response.